api/rest/cluster_bundle_handler.go

```diff
@@ -87,21 +94,29 @@ func (c *ClusterBundleHandler) Create(w http.ResponseWriter, r *http.Request) {
 		writeJSONError(w, http.StatusInsufficientStorage, fmt.Errorf("could not create data file %s: %s", id, err))
 		return
 	}
-	masters, err := c.tools.GetMasterNodes()
-	if err != nil {
-		if e := c.failed(bundle, err); e != nil {
-			logrus.WithField("ID", bundle.ID).Error(e.Error())
+
+	var masters, agents []dcos.Node
+
+	if options.Masters == nil || *options.Masters == true {
+		masters, err = c.tools.GetMasterNodes()
+		if err != nil {
+			if e := c.failed(bundle, err); e != nil {
+				logrus.WithField("ID", bundle.ID).Error(e.Error())
+			}
+			writeJSONError(w, http.StatusInternalServerError, fmt.Errorf("error getting master nodes for bundle %s: %s", id, err))
```

Comment:
As a general rule, I don't think we should be exposing internal errors up to the caller. An API's errors are part of our public contract, so should be more tightly controlled. I wouldn't say this is a blocker for this PR, since this is probably fairly rampant throughout our code, but it's something we should consider going forward. 

There are a lot of different approaches to this:

- https://middlemost.com/failure-is-your-domain/
- https://dzone.com/articles/rest-api-error-handling-problem-details-response

And even an RFC: https://tools.ietf.org/html/rfc7807

Reply:
> **Operator role**
> Finally, the last line of defense is the system operator which may be a developer or an operations person. These people understand the details of the system and can work with any kind of error.
> In this role, **you typically want to see as much information as possible**. In addition to the error code and human-readable message, a logical stack trace can help the operator understand the program flow.

So I think we are fine since this API is used only by operators and we need to give them all data they need to understand what's wrong.